### PR TITLE
Fix ipv6 parser

### PR DESCRIPTION
**Pull request by yakov-olkhovskiy**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fixed bug (Code: 632. DB::Exception: Unexpected data ... after parsed IPv6 value ...) with typed parsing of IP types from text source.

found here:
https://github.com/ClickHouse/ClickHouse/issues/45416#issuecomment-1387672077

### CODE REVIEW

**Comment by kitaisreal** [comment]
@yakov-olkhovskiy need to fix stateless tests issues.

**Review by kitaisreal — APPROVED** [review]

File: tests/queries/0_stateless/02118_deserialize_whole_text.sh [hunk 1]
```diff
@@ -33,10 +33,10 @@ echo "[\"255.255.255.255trash\"]" > $DATA_FILE
 $CLICKHOUSE_CLIENT -q "SELECT * FROM file('data_02118', 'JSONCompactStringsEachRow', 'x IPv4')" 2>&1 | grep -F -q "UNEXPECTED_DATA_AFTER_PARSED_VALUE" && echo 'OK' || echo 'FAIL'
 
 echo "255.255.255.255trash" > $DATA_FILE
-$CLICKHOUSE_CLIENT -q "SELECT * FROM file('data_02118', 'TSV', 'x IPv4')" 2>&1 | grep -F -q "UNEXPECTED_DATA_AFTER_PARSED_VALUE" && echo 'OK' || echo 'FAIL'
+$CLICKHOUSE_CLIENT -q "SELECT * FROM file('data_02118', 'TSV', 'x IPv4')" 2>&1 | grep -F -q "CANNOT_PARSE_INPUT_ASSERTION_FAILED" && echo 'OK' || echo 'FAIL'
```
Comment (kitaisreal) [inline tests/queries/0_stateless/02118_deserialize_whole_text.sh:36]: Why error code is not equal to `UNEXPECTED_DATA_AFTER_PARSED_VALUE` ?

**Review by kitaisreal — CHANGES_REQUESTED** [review]

File: tests/queries/0_stateless/02118_deserialize_whole_text.sh (hunk 1 above)
Comment (KochetovNicolai) [reply tests/queries/0_stateless/02118_deserialize_whole_text.sh:36]: I see that previously exception was from `SerializationIP` (which tried to parse a full string, and probably it lead to a bug from parser).
Now, exceptions are from `TabSeparatedFormatReader::skipRowEndDelimiter` (or other formats). Looks ok.

**Review by KochetovNicolai — APPROVED** [review]

File: tests/queries/0_stateless/02118_deserialize_whole_text.sh (hunk 1 above)
Comment (kitaisreal) [reply tests/queries/0_stateless/02118_deserialize_whole_text.sh:36]: Agreed.

**Review by kitaisreal — APPROVED** [review]

**Pull request merged**